sdk/resource/resource_test.go

```diff
@@ -452,6 +452,36 @@ func TestNew(t *testing.T) {
 	}
 }
 
+func TestWithHostID(t *testing.T) {
```

Comment:
How about adding a `TestWithHostIDSmoke`? 
The test would NOT mock host IR provider.
The test would assert if `res, err := resource.New(ctx, resource.WithHostID())` returns no error and that `host.id` attribute is not empty.

This way we will have a smoke test that could be executed on any OS.

Reply:
This HostID detection implemented in this PR is meant for non-containerized hosts. There are other rules for detecting HostID that are handled by cloud specific detectors. See the [GCE Detector](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/e7e73f6548c6f8f9114f647d8b154c7b0493c5a0/detectors/gcp/gce.go#L69) for example.

I would expect that the smoke test could fail in a containerized environment. We could try to add it and see how it goes in our build environment, but I worry it'd become a potentially flaky test, if it even works at all.

Reply:
Got it. Resolving.